cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/CustomerApiClient.java

```diff
@@ -48,15 +43,27 @@ public CustomerApiClient(HttpClient httpClient,
     @Override
     public Optional<CustomerResponse> getCustomer(String orgNumber) {
         logger.info("Requesting customer information for orgNumber: " + orgNumber);
-        return fetchCustomerInformation(orgNumber)
-            .stream()
-            .filter(responseIsSuccessful())
-            .map(tryParsingCustomer())
-            .findAny()
-            .flatMap(this::getValueOrLogError);
+        try {
```

Comment:
Hi. It looks OK, except for the error messages that could be constants.

However, if you want to simplify the handling of optionals and stuff you can refactor a bit. 

```
 @Override
    public Optional<CustomerResponse> getCustomer(String orgNumber) {
        logger.info("Requesting customer information for orgNumber: " + orgNumber);

        var response = fetchCustomerInformation(orgNumber);
        checkFoValidResponse(response);
        return Optional.ofNullable(parseCustomer(response));
    }

    private void checkFoValidResponse(HttpResponse<String> response) {
        if (responseIsUnexpected(response)) {
            logger.error("Error fetching customer information, API response was {}", response.statusCode());
            throw new IllegalStateException("Error fetching customer information");
        }
    }

    private boolean responseIsUnexpected(HttpResponse<String> response) {
        return response.statusCode() != HttpStatus.SC_OK && response.statusCode() != HttpStatus.SC_NOT_FOUND;
    }

    private IllegalStateException getHttpClientInitializationError() {
        logger.error("Error fetching customer information");
        return new IllegalStateException("Error fetching customer information, http client failed to initialize.");
    }

    private HttpResponse<String> fetchCustomerInformation(String orgNumber) {
        return Try.of(formUri(orgNumber))
            .map(URIBuilder::build)
            .map(this::buildHttpRequest)
            .map(this::sendHttpRequest)
            .orElseThrow(fail -> getHttpClientInitializationError());
    }

    private CustomerResponse parseCustomer(HttpResponse<String> response) {
        if (response.statusCode() == HttpStatus.SC_NOT_FOUND) {
            return null;
        }
        return attempt(() -> objectMapper.readValue(response.body(), CustomerResponse.class))
            .orElseThrow(fail -> handleJsonProcessingError(fail.getException()));
    }

    private IllegalStateException handleJsonProcessingError(Exception e) {
        logger.error("Error parsing customer information", e);
        return new IllegalStateException("Error parsing customer information");
    }
```